packages/events-example/src/commands/reconcile/index.ts

```diff
@@ -1,84 +1,117 @@
-import { Command, Flags } from '@oclif/core';
+import { Command } from '@oclif/core';
 import Nile, { Instance, NileApi } from '@theniledev/js';
 
-import { pulumiProgram } from '../../pulumiS3';
-import PulumiAwsDeployment from '../../deployments/PulumiAwsDeployment';
+import { pulumiS3, PulumiAwsDeployment } from './lib/pulumi';
 import { ReconciliationPlan } from './ReconciliationPlan';
+import { flagDefaults } from './flagDefaults';
+
+// configuration for interacting with nile
+type NileConfig = {
+  workspace: string; // workspace of the application
+  basePath: string; // nile URL (https://prod.thenile.dev)
+};
+type DeveloperCreds = {
+  email: string; // developer email
+  password: string; // developer password
+};
 
 export default class Reconcile extends Command {
   static enableJsonFlag = true;
   static description = 'reconcile nile/pulumi deploys';
-
-  static flags = {
-    basePath: Flags.string({
-      description: 'root URL for the Nile API',
-      default: 'http://localhost:8080',
-    }),
-    workspace: Flags.string({
-      description: 'your Nile workspace name',
-      default: 'dev',
-    }),
-    email: Flags.string({
-      description: 'developer email address',
-      default: 'developer@demo.com',
-    }),
-    password: Flags.string({
-      description: 'developer password',
-      default: 'very_secret',
-    }),
-    organization: Flags.string({ description: 'an organization in your Nile workspace' }),
-    entity: Flags.string({ description: 'an entity type in your Nile workspace' }),
-    status: Flags.boolean({ char: 's', description: 'check current status of your control and data planes', default: false }),
-    region: Flags.string({ description: 'AWS region', default: 'us-west-2'}),
-  };
+  static flags = flagDefaults;
 
   deployment!: PulumiAwsDeployment;
   nile!: NileApi;
 
-  async run(): Promise<any> {
+  /**
+   * Runner for oclif
+   * @returns void
+   */
+  async run(): Promise<unknown> {
     const { flags } = await this.parse(Reconcile);
 
-    // nile setup
-    await this.connectNile(flags);
-
-    // pulumi setup
-    this.deployment = await PulumiAwsDeployment.create("nile-examples", pulumiProgram, { region: flags.region });
-
-    // load our data
-    const stacks = await this.deployment.loadPulumiStacks();
-    const instances = await this.loadNileInstances(
-      flags.organization,
-      flags.entity
+    const {
```

Comment:
I hadn't quite finished cleaning this up, so there's going to be some conflicts with #411.